prov/efa/src/rxr/rxr_pkt_cmd.c

```diff
@@ -1055,6 +1055,116 @@ void rxr_pkt_proc_received(struct rxr_ep *ep, struct rxr_pkt_entry *pkt_entry)
 	}
 }
 
+/**
+ * @brief Read peer raw address from packet header and insert the peer in AV.
+ * The caller of this function must guarantee that the packet header contain the raw address.
+ * @param ep Pointer to RDM endpoint
+ * @param pkt_entry Pointer to packet entry
+ * @returns Peer address
+ */
+static inline fi_addr_t rxr_pkt_determine_addr_from_hdr(struct rxr_ep *ep, struct rxr_pkt_entry *pkt_entry)
+{
+	void *raw_addr;
+
+	raw_addr = rxr_pkt_req_raw_addr(pkt_entry);
+	assert(raw_addr);
+	return rxr_pkt_insert_addr(ep, pkt_entry, raw_addr);
+};
+
+
+#if HAVE_EFADV_CQ_EX
+/**
+ * @brief Read peer raw address from EFA device and look up the peer address in AV.
+ * The caller must guarantee that the peer AH is unknown.
+ * @return Peer address, or FI_ADDR_NOTAVAIL if unavailable.
+ */
+static inline fi_addr_t rxr_pkt_determine_address_from_efadv(struct rxr_ep *ep, struct rxr_pkt_entry *pkt_entry)
+{
+	struct efa_cq *efa_cq;
+	struct efa_ep *efa_ep;
+	struct efa_ep_addr efa_ep_addr = {0};
+	fi_addr_t addr;
+	union ibv_gid gid = {0};
+	uint32_t *connid = NULL;
+
+	efa_cq = container_of(ep->rdm_cq, struct efa_cq, util_cq.cq_fid);
+
+	/* Attempt to read AH from EFA firmware */
+	if (efadv_wc_read_ah(efadv_cq_from_ibv_cq_ex(efa_cq->ibv_cq_ex), &gid) >= 0) {
+		/* AH should be unknown (negative) */
+		addr = FI_ADDR_NOTAVAIL;
+		goto out;
```

Comment:
`return FI_ADDR_NOTAVAIL` here should be good. `goto` is used as the last resort when there are lots of resources to release.

Reply:
Sure will address.